lib/generators/rspec/integration/templates/api_request_spec.rb

```diff
@@ -11,59 +11,65 @@
 # It only uses APIs available in rails and/or rspec-rails.  There are a number
 # of tools you can use to make these specs even more expressive, but we're
 # sticking to rails and rspec-rails APIs to keep things simple and stable.
-#
-# Compared to earlier versions of this generator, there is very limited use of
-# stubs and message expectations in this spec.  Stubs are only used when there
-# is no simpler way to get a handle on the object needed for the example.
-# Message expectations are only used when there is no simpler way to specify
-# that an instance is receiving a specific message.
-#
-# Also compared to earlier versions of this generator, there are no longer any
-# expectations of assigns and templates rendered. These features have been
-# removed from Rails core in Rails 5, but can be added back in via the
-# `rails-controller-testing` gem.
+
 
 <% module_namespacing do -%>
-RSpec.describe <%= controller_class_name %>Controller, <%= type_metatag(:controller) %> do
+RSpec.describe "<%= class_name.pluralize %>", <%= type_metatag(:request) %> do
 
   # This should return the minimal set of attributes required to create a valid
   # <%= class_name %>. As you add validations to <%= class_name %>, be sure to
   # adjust the attributes here as well.
-  let(:valid_attributes) {
+  let(:valid_attributes) do
+<% if options[:factorybot] || Gem.loaded_specs.has_key?('factory_bot_rails') -%>
+     attributes_for(:<%= ns_file_name -%>)
+<% elsif options[:fabrication] -%>
+    Fabricate.attributes_for(<%= ns_prefix.empty? ? ":"+ns_file_name : "'#{class_name}'"-%>)
+<% else -%>
     skip("Add a hash of attributes valid for your model")
-  }
+<% end -%>
+  end
 
-  let(:invalid_attributes) {
+  let(:invalid_attributes) do
     skip("Add a hash of attributes invalid for your model")
-  }
+  end
 
-  # This should return the minimal set of values that should be in the session
+  # This should return the minimal set of values that should be in the headers
   # in order to pass any filters (e.g. authentication) defined in
-  # <%= controller_class_name %>Controller. Be sure to keep this updated too.
-  let(:valid_session) { {} }
+  # <%= class_name %>Controller. Be sure to keep this updated too.
+  # Because it's an API request a default JSON header is added.
+  # If you have an Authentication - and you definitely should - add your token header here.
+  let(:valid_headers) do
+    {
+      "Content-Type" => "application/json"
+    }
+  end
 
 <% unless options[:singleton] -%>
   describe "GET #index" do
     it "returns a success response" do
       <%= file_name %> = <%= class_name %>.create! valid_attributes
-<% if RUBY_VERSION < '1.9.3' -%>
-      get :index, {}, valid_session
-<% else -%>
-      get :index, params: {}, session: valid_session
-<% end -%>
```

Comment:
This is necessary for older Rails / Ruby, your new file will have to cater for all supported versions of Rails, this is will be easier in `4-0-dev` / once `rspec-rails` 4 is released as we target only 5.0 upwards.

Reply:
This can now safely be ignored, you can use rails 5 syntax.